### PR TITLE
fix: #114

### LINKED ISSUE #114: docker build fails
**Affected Component**
<!-- Please tell us which component of GCP Scanner is affected (leave empty if you are unsure). -->

* [x] Docker build

**Describe the bug**
<!-- Please provide a clear and concise description of what the bug is. Code samples should be put in the **To Reproduce** section. -->

This is a known issue with `Python3.7`
https://github.com/grpc/grpc/issues/24556
```
=> [internal] load build context                                                                                                                                                                      0.3s
 => => transferring context: 93.91kB                                                                                                                                                                   0.0s
 => [2/8] RUN mkdir /home/sa_scanner                                                                                                                                                                   0.4s
 => [3/8] COPY src/ /home/sa_scanner/                                                                                                                                                                  0.0s
 => [4/8] COPY pyproject.toml /home/sa_scanner/                                                                                                                                                        0.0s
 => [5/8] COPY README.md /home/sa_scanner                                                                                                                                                              0.0s
 => [6/8] WORKDIR /home/sa_scanner                                                                                                                                                                     0.0s
 => ERROR [7/8] RUN pip install .                                                                                                                                                                     18.9s
------                                                                                                                                                                                                      
 > [7/8] RUN pip install .:                                                                                                                                                                                 
#12 1.139 Processing /home/sa_scanner                                                                                                                                                                       
#12 1.141   Installing build dependencies: started                                                                                                                                                          
#12 5.902   Installing build dependencies: finished with status 'done'                                                                                                                                      
#12 5.905   Getting requirements to build wheel: started                                                                                                                                                    
#12 5.967   Getting requirements to build wheel: finished with status 'done'
#12 5.972   Preparing metadata (pyproject.toml): started
#12 6.100   Preparing metadata (pyproject.toml): finished with status 'done'
#12 6.500 Collecting google-api-python-client==2.80.0
#12 6.662   Downloading google_api_python_client-2.80.0-py2.py3-none-any.whl (11.0 MB)
#12 9.259      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 11.0/11.0 MB 4.3 MB/s eta 0:00:00
#12 9.344 Collecting google-cloud-container==2.17.4
#12 9.391   Downloading google_cloud_container-2.17.4-py2.py3-none-any.whl (217 kB)
#12 9.441      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 217.7/217.7 kB 4.4 MB/s eta 0:00:00
#12 9.580 Collecting google-cloud-iam==2.11.2
#12 9.630   Downloading google_cloud_iam-2.11.2-py2.py3-none-any.whl (115 kB)
#12 9.652      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 115.0/115.0 kB 5.3 MB/s eta 0:00:00
#12 9.718 Collecting httplib2==0.21.0
#12 9.760   Downloading httplib2-0.21.0-py3-none-any.whl (96 kB)
#12 9.778      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 96.8/96.8 kB 5.5 MB/s eta 0:00:00
#12 9.825 Collecting pyu2f==0.1.5
#12 9.874   Downloading pyu2f-0.1.5.tar.gz (27 kB)
#12 9.885   Preparing metadata (setup.py): started
#12 10.28   Preparing metadata (setup.py): finished with status 'done'
#12 10.38 Collecting requests==2.28.2
#12 10.48   Downloading requests-2.28.2-py3-none-any.whl (62 kB)
#12 10.49      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 62.8/62.8 kB 4.8 MB/s eta 0:00:00
#12 10.65 Collecting google-auth<3.0.0dev,>=1.19.0
#12 10.70   Downloading google_auth-2.16.2-py2.py3-none-any.whl (177 kB)
#12 10.75      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 177.2/177.2 kB 4.1 MB/s eta 0:00:00
#12 10.80 Collecting google-auth-httplib2>=0.1.0
#12 10.84   Downloading google_auth_httplib2-0.1.0-py2.py3-none-any.whl (9.3 kB)
#12 10.93 Collecting google-api-core!=2.0.*,!=2.1.*,!=2.2.*,!=2.3.0,<3.0.0dev,>=1.31.5
#12 10.98   Downloading google_api_core-2.11.0-py3-none-any.whl (120 kB)
#12 11.01      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 120.3/120.3 kB 4.2 MB/s eta 0:00:00
#12 11.11 Collecting uritemplate<5,>=3.0.1
#12 11.19   Downloading uritemplate-4.1.1-py2.py3-none-any.whl (10 kB)
#12 11.31 Collecting proto-plus<2.0.0dev,>=1.22.0
#12 11.40   Downloading proto_plus-1.22.2-py3-none-any.whl (47 kB)
#12 11.49      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 47.9/47.9 kB 6.2 MB/s eta 0:00:00
#12 11.79 Collecting protobuf!=3.20.0,!=3.20.1,!=4.21.0,!=4.21.1,!=4.21.2,!=4.21.3,!=4.21.4,!=4.21.5,<5.0.0dev,>=3.19.5
#12 11.83   Downloading protobuf-4.22.1-cp37-abi3-manylinux2014_aarch64.whl (301 kB)
#12 11.90      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 301.2/301.2 kB 4.7 MB/s eta 0:00:00
#12 11.98 Collecting pyparsing!=3.0.0,!=3.0.1,!=3.0.2,!=3.0.3,<4,>=2.4.2
#12 12.03   Downloading pyparsing-3.0.9-py3-none-any.whl (98 kB)
#12 12.04      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 98.3/98.3 kB 8.0 MB/s eta 0:00:00
#12 12.10 Collecting six
#12 12.14   Downloading six-1.16.0-py2.py3-none-any.whl (11 kB)
#12 12.25 Collecting charset-normalizer<4,>=2
#12 12.29   Downloading charset_normalizer-3.1.0-cp311-cp311-manylinux_2_17_aarch64.manylinux2014_aarch64.whl (193 kB)
#12 12.32      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 194.0/194.0 kB 6.9 MB/s eta 0:00:00
#12 12.38 Collecting idna<4,>=2.5
#12 12.42   Downloading idna-3.4-py3-none-any.whl (61 kB)
#12 12.43      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 61.5/61.5 kB 6.6 MB/s eta 0:00:00
#12 12.50 Collecting urllib3<1.27,>=1.21.1
#12 12.55   Downloading urllib3-1.26.15-py2.py3-none-any.whl (140 kB)
#12 12.57      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 140.9/140.9 kB 7.0 MB/s eta 0:00:00
#12 12.63 Collecting certifi>=2017.4.17
#12 12.68   Downloading certifi-2022.12.7-py3-none-any.whl (155 kB)
#12 12.70      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 155.3/155.3 kB 7.5 MB/s eta 0:00:00
#12 12.76 Collecting googleapis-common-protos<2.0dev,>=1.56.2
#12 12.80   Downloading googleapis_common_protos-1.59.0-py2.py3-none-any.whl (223 kB)
#12 12.83      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 223.6/223.6 kB 8.1 MB/s eta 0:00:00
#12 13.46 Collecting grpcio<2.0dev,>=1.33.2
#12 13.51   Downloading grpcio-1.51.3.tar.gz (22.1 MB)
#12 16.95      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 22.1/22.1 MB 6.9 MB/s eta 0:00:00
#12 18.13   Preparing metadata (setup.py): started
#12 18.38   Preparing metadata (setup.py): finished with status 'error'
#12 18.38   error: subprocess-exited-with-error
#12 18.38   
#12 18.38   × python setup.py egg_info did not run successfully.
#12 18.38   │ exit code: 1
#12 18.38   ╰─> [14 lines of output]
#12 18.38       Traceback (most recent call last):
#12 18.38         File "<string>", line 2, in <module>
#12 18.38         File "<pip-setuptools-caller>", line 34, in <module>
#12 18.38         File "/tmp/pip-install-7egp0l84/grpcio_2bc3ade628c0443a83519a9e2c691087/setup.py", line 262, in <module>
#12 18.38           if check_linker_need_libatomic():
#12 18.38              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
#12 18.38         File "/tmp/pip-install-7egp0l84/grpcio_2bc3ade628c0443a83519a9e2c691087/setup.py", line 209, in check_linker_need_libatomic
#12 18.38           cpp_test = subprocess.Popen(cxx + ['-x', 'c++', '-std=c++14', '-'],
#12 18.38                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
#12 18.38         File "/usr/local/lib/python3.11/subprocess.py", line 1024, in __init__
#12 18.38           self._execute_child(args, executable, preexec_fn, close_fds,
#12 18.38         File "/usr/local/lib/python3.11/subprocess.py", line 1901, in _execute_child
#12 18.38           raise child_exception_type(errno_num, err_msg, err_filename)
#12 18.38       FileNotFoundError: [Errno 2] No such file or directory: 'c++'
#12 18.38       [end of output]
#12 18.38   
#12 18.38   note: This error originates from a subprocess, and is likely not a problem with pip.
#12 18.38 error: metadata-generation-failed
#12 18.38 
#12 18.38 × Encountered error while generating package metadata.
#12 18.38 ╰─> See above for output.
#12 18.38 
#12 18.38 note: This is an issue with the package mentioned above, not pip.
#12 18.38 hint: See above for details.
#12 18.40 
#12 18.40 [notice] A new release of pip available: 22.3.1 -> 23.0.1
#12 18.40 [notice] To update, run: pip install --upgrade pip
------
executor failed running [/bin/sh -c pip install .]: exit code: 1
```

**To Reproduce**

`docker build -t gcp_scan .`


I am willing to fix it and open a PR :)

### PR DESCRIPTION
**Pull request by grumpyp**
This PR fixes #114 

All information can be read in the issue.

### CODE REVIEW

**Comment by grumpyp** [comment]
Thanks for the hint. Done

File: Dockerfile [hunk 1]
```diff
@@ -23,5 +23,4 @@ COPY README.md /home/sa_scanner
 WORKDIR /home/sa_scanner
 RUN pip install .
 
-WORKDIR /home/sa_scanner/
```
Comment (ZetaTwo) [inline Dockerfile:26]: Why is this line removed?

File: Dockerfile [hunk 2]
```diff
@@ -13,7 +13,7 @@
 # limitations under the License.
 
 
-FROM python:3-slim-buster
+FROM python:3.8-slim-buster
```
Comment (ZetaTwo) [inline Dockerfile:16]: According to https://hub.docker.com/_/python, `3-slim-buster` corresponds to python 3.11 but now we are downgrading to 3.8, why?

Comment (grumpyp) [reply Dockerfile:16]: it corresponds to `python3.7` from what I know. So I upgraded to `Python3.8`

File: Dockerfile (hunk 1 above)
Comment (grumpyp) [reply Dockerfile:26]: cause it's obsolete

File: Dockerfile (hunk 2 above)
Comment (ZetaTwo) [reply Dockerfile:16]: If you check the page I linked you can see that these are all the same images: 3.11.2-slim-buster, 3.11-slim-buster, 3-slim-buster, slim-buster

File: Dockerfile (hunk 1 above)
Comment (ZetaTwo) [reply Dockerfile:26]: Could you clarify this, the docs recommend using it: https://docs.docker.com/engine/reference/builder/#:~:text=Therefore%2C%20to%20avoid%20unintended%20operations%20in%20unknown%20directories%2C%20it%20is%20best%20practice%20to%20set%20your%20WORKDIR%20explicitly.

File: Dockerfile [hunk 3]
```diff
@@ -23,5 +23,4 @@ COPY README.md /home/sa_scanner
 WORKDIR /home/sa_scanner
```
Comment (grumpyp) [inline Dockerfile:23]: It's set here already @ZetaTwo

Comment (ZetaTwo) [reply Dockerfile:23]: Ok, then it makes sense. Your initial comment was quite confusing.

File: Dockerfile (hunk 2 above)
Comment (ZetaTwo) [reply Dockerfile:16]: Can you instead try adding a `RUN apt-get install -y g++` to the dockerfile to see if that solves it?

Comment (grumpyp) [reply Dockerfile:16]: I am afraid, it does output the same error:

```
...
#12 6.048 Collecting grpcio<2.0dev,>=1.33.2
#12 6.086   Downloading grpcio-1.51.3.tar.gz (22.1 MB)
#12 8.259      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 22.1/22.1 MB 10.1 MB/s eta 0:00:00
#12 9.315   Preparing metadata (setup.py): started
#12 9.628   Preparing metadata (setup.py): finished with status 'error'
#12 9.635   error: subprocess-exited-with-error
#12 9.635   
#12 9.635   × python setup.py egg_info did not run successfully.
#12 9.635   │ exit code: 1
#12 9.635   ╰─> [14 lines of output]
#12 9.635       Traceback (most recent call last):
#12 9.635         File "<string>", line 2, in <module>
#12 9.635         File "<pip-setuptools-caller>", line 34, in <module>
#12 9.635         File "/tmp/pip-install-99ma6hvk/grpcio_0a81b05cdcf9441c808fc788f2c54d42/setup.py", line 262, in <module>
#12 9.635           if check_linker_need_libatomic():
#12 9.635              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
#12 9.635         File "/tmp/pip-install-99ma6hvk/grpcio_0a81b05cdcf9441c808fc788f2c54d42/setup.py", line 209, in check_linker_need_libatomic
#12 9.635           cpp_test = subprocess.Popen(cxx + ['-x', 'c++', '-std=c++14', '-'],
#12 9.635                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
#12 9.635         File "/usr/local/lib/python3.11/subprocess.py", line 1024, in __init__
#12 9.635           self._execute_child(args, executable, preexec_fn, close_fds,
#12 9.635         File "/usr/local/lib/python3.11/subprocess.py", line 1901, in _execute_child
#12 9.635           raise child_exception_type(errno_num, err_msg, err_filename)
#12 9.635       FileNotFoundError: [Errno 2] No such file or directory: 'c++'
#12 9.635       [end of output]
#12 9.635   
#12 9.635   note: This error originates from a subprocess, and is likely not a problem with pip.
#12 9.637 error: metadata-generation-failed
#12 9.637 
#12 9.637 × Encountered error while generating package metadata.
#12 9.637 ╰─> See above for output.
#12 9.637 
#12 9.637 note: This is an issue with the package mentioned above, not pip.
#12 9.637 hint: See above for details.
#12 9.653 
#12 9.653 [notice] A new release of pip available: 22.3.1 -> 23.0.1
#12 9.653 [notice] To update, run: pip install --upgrade pip
------
executor failed running [/bin/sh -c pip install .]: exit code: 1
```

using this

```

FROM python:3-slim-buster

RUN mkdir /home/sa_scanner
COPY src/ /home/sa_scanner/
COPY pyproject.toml /home/sa_scanner/
COPY README.md /home/sa_scanner

WORKDIR /home/sa_scanner
RUN apt-get install -y g++

RUN pip install .


ENTRYPOINT ["gcp-scanner"]
```

Comment (ZetaTwo) [reply Dockerfile:16]: What about `RUN apt-get install -y build-essential` instead? I'm really not a fan of locking down to a fairly old Python version as we want to test that everything works on up-to-date Python versions.

Comment (grumpyp) [reply Dockerfile:16]: I tried it, which led to an error so I inserted an `RUN apt-get update` to make it work. It ended in an endless wheel
```
 => [7/9] RUN apt-get update                                                                                                                      2.7s
 => [8/9] RUN apt-get install -y build-essential                                                                                                 11.9s
 => [9/9] RUN pip install .                                                                                                                     130.7s 
 => => #   Building wheel for pyu2f (setup.py): finished with status 'done'                                                                            
 => => #   Created wheel for pyu2f: filename=pyu2f-0.1.5-py3-none-any.whl size=39401 sha256=1dbc3d3135c35f20bcf4ccfdd4c0823ec54aae36ce2b298782f28265ce 
 => => # c3a2d3                                                                                                                                        
 => => #   Stored in directory: /root/.cache/pip/wheels/a0/84/f9/1676c313d1bf6c4c30f783a7c5b18648622d5f1b2f4caf468a                                    
 => => #   Building wheel for grpcio (setup.py): started                                                                                               
 => => #   Building wheel for grpcio (setup.py): still running...   
 ```

I also tried 
```
pip install --upgrade pip
pip install --upgrade setuptools wheel
```
which also didn't work.

I suggest using `python:3.10-bullseye`. It works and the version wouldn't be so old. What do you think?

File: Dockerfile [hunk 4]
```diff
@@ -13,7 +13,7 @@
 # limitations under the License.
 
 
-FROM python:3-slim-buster
+FROM python:3.10-bullseye
```
Comment (ZetaTwo) [inline Dockerfile:16]: It seems like you switched to a different branch completely. In that case can you please test with just `python:bullseye`

**Review by ZetaTwo — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 4 above)
Comment (grumpyp) [reply Dockerfile]: Hi @ZetaTwo 

It seems like this is the solution.

```
[+] Building 50.7s (13/13) FINISHED                                                                                                                             
 => [internal] load build definition from Dockerfile                                                                                                       0.0s
 => => transferring dockerfile: 841B                                                                                                                       0.0s
 => [internal] load .dockerignore                                                                                                                          0.0s
 => => transferring context: 2B                                                                                                                            0.0s
 => [internal] load metadata for docker.io/library/python:bullseye                                                                                         2.1s
 => [auth] library/python:pull token for registry-1.docker.io                                                                                              0.0s
 => [1/7] FROM docker.io/library/python:bullseye@sha256:21ce92a075cf9c454a936f925e058b4d8fc0cfc7a05b9e877bed4687c51a5658                                  35.9s
 => => resolve docker.io/library/python:bullseye@sha256:21ce92a075cf9c454a936f925e058b4d8fc0cfc7a05b9e877bed4687c51a5658                                   0.0s
 => => sha256:9c5aa3f126bd652d22fcb5fbaea0446609747aa71454fd01a07810b6172e4b78 2.22kB / 2.22kB                                                             0.0s
 => => sha256:8022b074731d9ecee7f4fba79b993920973811dda168bbc08636f18523b90122 53.70MB / 53.70MB                                                          13.1s
 => => sha256:21ce92a075cf9c454a936f925e058b4d8fc0cfc7a05b9e877bed4687c51a5658 1.65kB / 1.65kB                                                             0.0s
 => => sha256:74707af2d4ec537fd45394f8f6a9d93b1cd6b939f2dc942c2c09cdfa1e636102 7.92kB / 7.92kB                                                             0.0s
 => => sha256:7971239fe1d69763272ccc0b2527efa95547d37c53630ed0a71db4e00d3ef964 5.15MB / 5.15MB                                                             0.7s
 => => sha256:26c861b53509d61c37240d2f80efb3a351d2f1d7f4f8e8ec2e5004c1d86af89c 10.87MB / 10.87MB                                                           2.7s
 => => sha256:1714880ecc1c021a5f708f4369f91d3c2c53b998a56d563d0a9aa9be2488d794 54.68MB / 54.68MB                                                          19.4s
 => => sha256:895a945a1f9ba441c2748501c4d46569edfbc2bfbdb9b47d41e753e752247fdc 189.73MB / 189.73MB                                                        28.1s
 => => sha256:2a83fe9e3053a5f52e699f1078c2abeaad849bb81da9068f1aa541bb5673a21f 6.40MB / 6.40MB                                                            14.7s
 => => extracting sha256:8022b074731d9ecee7f4fba79b993920973811dda168bbc08636f18523b90122                                                                  1.6s
 => => sha256:cad84c97de70ce8ac18eaca0f13ed88dd622a1ee17d4f47c88941116b83a8cac 19.93MB / 19.93MB                                                          20.0s
 => => extracting sha256:7971239fe1d69763272ccc0b2527efa95547d37c53630ed0a71db4e00d3ef964                                                                  0.2s
 => => extracting sha256:26c861b53509d61c37240d2f80efb3a351d2f1d7f4f8e8ec2e5004c1d86af89c                                                                  0.2s
 => => sha256:317609cadfd3e8a9eb5393f8aae884de904c93391615b2eb841f17449709e1b5 243B / 243B                                                                19.6s
 => => extracting sha256:1714880ecc1c021a5f708f4369f91d3c2c53b998a56d563d0a9aa9be2488d794                                                                  2.3s
 => => sha256:f756083c502e7d8ae0753f9c828624027c0b16b8d4a203c46929e9606b459ab9 3.08MB / 3.08MB                                                            20.8s
 => => extracting sha256:895a945a1f9ba441c2748501c4d46569edfbc2bfbdb9b47d41e753e752247fdc                                                                  5.7s
 => => extracting sha256:2a83fe9e3053a5f52e699f1078c2abeaad849bb81da9068f1aa541bb5673a21f                                                                  0.2s
 => => extracting sha256:cad84c97de70ce8ac18eaca0f13ed88dd622a1ee17d4f47c88941116b83a8cac                                                                  0.9s
 => => extracting sha256:317609cadfd3e8a9eb5393f8aae884de904c93391615b2eb841f17449709e1b5                                                                  0.0s
 => => extracting sha256:f756083c502e7d8ae0753f9c828624027c0b16b8d4a203c46929e9606b459ab9                                                                  0.2s
 => [internal] load build context                                                                                                                          0.2s
 => => transferring context: 546B                                                                                                                          0.0s
 => [2/7] RUN mkdir /home/sa_scanner                                                                                                                       0.4s
 => [3/7] COPY src/ /home/sa_scanner/                                                                                                                      0.1s
 => [4/7] COPY pyproject.toml /home/sa_scanner/                                                                                                            0.0s
 => [5/7] COPY README.md /home/sa_scanner                                                                                                                  0.0s
 => [6/7] WORKDIR /home/sa_scanner                                                                                                                         0.0s
 => [7/7] RUN pip install .                                                                                                                               11.5s
 => exporting to image                                                                                                                                     0.6s
 => => exporting layers                                                                                                                                    0.6s
 => => writing image sha256:e07896f3533ac8a38bf573d9d8d080033d3346b89fec57a04020eeaba4b6e9eb                                                               0.0s
 => => naming to docker.io/library/gcptest                                                                                                                 0.0
 ```
 
 Once it's merged I close the issue of our re-consideration of version locking.

Comment (ZetaTwo) [reply Dockerfile]: That's great. Unfortunately, I need to ask you to do one final test. Can you test if `python:slim-bullseye` works?

Comment (grumpyp) [reply Dockerfile]:
```
patrickgerard ~/Documents/Github/gcp_scanner [main] $ docker build . -t gcppt
[+] Building 19.3s (13/13) FINISHED                                                                                                                              
 => [internal] load build definition from Dockerfile                                                                                                        0.0s
 => => transferring dockerfile: 846B                                                                                                                        0.0s
 => [internal] load .dockerignore                                                                                                                           0.0s
 => => transferring context: 2B                                                                                                                             0.0s
 => [internal] load metadata for docker.io/library/python:slim-bullseye                                                                                     2.0s
 => [auth] library/python:pull token for registry-1.docker.io                                                                                               0.0s
 => [1/7] FROM docker.io/library/python:slim-bullseye@sha256:2f749ef90f54fd4b3c77cde78eec23ab5b8199d9ac84e4ced6ae523ef223ef7b                               6.3s
 => => resolve docker.io/library/python:slim-bullseye@sha256:2f749ef90f54fd4b3c77cde78eec23ab5b8199d9ac84e4ced6ae523ef223ef7b                               0.0s
 => => sha256:474f1e977c7dcc3c47ee9fee5d50764763c111f7ce8a0ce2282638fdf6135919 1.37kB / 1.37kB                                                              0.0s
 => => sha256:2704c6131c1c48593359f62312beb65f1173f8cfe8815c92e4fbdc97b493c8dd 6.91kB / 6.91kB                                                              0.0s
 => => sha256:fcdb9667c46b09d1c1d058681ea4a1db41e66bbc1a71d873a0c9da4f7a92947d 30.06MB / 30.06MB                                                            4.2s
 => => sha256:93a67280259393dc61b3b00d448dfe8c71e2fa81bfd4242f464ee813460608c7 1.06MB / 1.06MB                                                              1.0s
 => => sha256:163225f81387d77f1fdf0aa8292f0612e3754ae1039fc52cf73618f7e21cff3e 12.02MB / 12.02MB                                                            3.4s
 => => sha256:2f749ef90f54fd4b3c77cde78eec23ab5b8199d9ac84e4ced6ae523ef223ef7b 1.65kB / 1.65kB                                                              0.0s
 => => sha256:791f7f0a4fe70aa88ffb61f451ba1b46937c04b104e52a260f85be262e906e83 244B / 244B                                                                  1.5s
 => => sha256:e1fcd7e71152298be00c19f748bc3716f84cf31c472cb19b313c6a9b2bd4cbc9 3.37MB / 3.37MB                                                              2.9s
 => => extracting sha256:fcdb9667c46b09d1c1d058681ea4a1db41e66bbc1a71d873a0c9da4f7a92947d                                                                   1.2s
 => => extracting sha256:93a67280259393dc61b3b00d448dfe8c71e2fa81bfd4242f464ee813460608c7                                                                   0.1s
 => => extracting sha256:163225f81387d77f1fdf0aa8292f0612e3754ae1039fc52cf73618f7e21cff3e                                                                   0.3s
 => => extracting sha256:791f7f0a4fe70aa88ffb61f451ba1b46937c04b104e52a260f85be262e906e83                                                                   0.0s
 => => extracting sha256:e1fcd7e71152298be00c19f748bc3716f84cf31c472cb19b313c6a9b2bd4cbc9                                                                   0.1s
 => [internal] load build context                                                                                                                           0.1s
 => => transferring context: 546B                                                                                                                           0.0s
 => [2/7] RUN mkdir /home/sa_scanner                                                                                                                        0.4s
 => [3/7] COPY src/ /home/sa_scanner/                                                                                                                       0.0s
 => [4/7] COPY pyproject.toml /home/sa_scanner/                                                                                                             0.0s
 => [5/7] COPY README.md /home/sa_scanner                                                                                                                   0.0s
 => [6/7] WORKDIR /home/sa_scanner                                                                                                                          0.0s
 => [7/7] RUN pip install .                                                                                                                                 9.7s
 => exporting to image                                                                                                                                      0.6s
 => => exporting layers                                                                                                                                     0.6s
 => => writing image sha256:86853b95e16c46f82e0c0bed5a1b7b8fdcca0545af27116ddd23dbcd4d73f490                                                                0.0s
 => => naming to docker.io/library/gcppt                                                                                                                    0.0s
```

optimized :)

**Review by ZetaTwo — APPROVED** [review]
